idprovider/idprovidertest/idprovider.go

```diff
@@ -31,34 +31,34 @@ import (
 // NewYAMLFileT is the test friendly version of NewYAMLFile.
 // It uses the passed testing.T to handle the errors and registers the cleanup functions on it.
 func NewYAMLFileT(t *testing.T, peers ...perun.Peer) string {
```

Comment:
```suggestion
func NewPeerIDProviderT(t *testing.T, peers ...perun.Peer) string {
```

As discussed in other comments,the file format used 'yaml' in this case should be only an internal implementation detail and should not be included when naming public symbols of this package.

Also update it in the other locations in this file.